packages/wmr/src/lib/transform-prefresh-registrations.js

```diff
@@ -0,0 +1,409 @@
+				// 		continue;
+				// 	}
+				// 	const refParent = ref.parent;
+				// 	if (refParent.type === 'JSXOpeningElement') {
+				// 		isLikelyUsedAsType = true;
+				// 	} else if (refParent.type === 'CallExpression') {
+				// 		const callee = refParent.callee;
+				// 		let fnName;
+				// 		switch (callee.type) {
+				// 			case 'Identifier':
+				// 				fnName = callee.name;
+				// 				break;
+				// 			case 'MemberExpression':
+				// 				fnName = callee.property.name;
+				// 				break;
+				// 		}
+				// 		switch (fnName) {
+				// 			case 'createElement':
+				// 			case 'jsx':
+				// 			case 'jsxDEV':
+				// 			case 'jsxs':
+				// 				isLikelyUsedAsType = true;
+				// 				break;
+				// 		}
+				// 	}
+				// 	if (isLikelyUsedAsType) {
+				// 		// const X = ... + later <X />
+				// 		callback(inferredName, init, initPath);
+				// 		return true;
+				// 	}
+				// }
+			}
+		}
+		return false;
+	}
+
+	// const createContextTemplate = template(
+	// 	`
+	//   Object.assign((CREATECONTEXT.IDENT || (CREATECONTEXT.IDENT=CREATECONTEXT(VALUE))), {__:VALUE});
+	// `,
+	// 	{ placeholderPattern: /^[A-Z]+$/ }
+	// );
+
+	// const emptyTemplate = template(`
+	//   (CREATECONTEXT.IDENT || (CREATECONTEXT.IDENT=CREATECONTEXT()));
+	// `);
+
+	// const getFirstNonTsExpression = expression =>
+	// 	expression.type === 'TSAsExpression' ? getFirstNonTsExpression(expression.expression) : expression;
+
+	return {
+		name: 'transform-prefresh-registrations',
+		visitor: {
+			ClassDeclaration: {
+				enter(path) {
+					const node = path.node;
+					let programPath;
+					let insertAfterPath;
+					switch (path.parent.type) {
+						case 'Program':
+							insertAfterPath = path;
+							programPath = path.parentPath;
+							break;
+						case 'ExportNamedDeclaration':
+							insertAfterPath = path.parentPath;
+							programPath = insertAfterPath.parentPath;
+							break;
+						case 'ExportDefaultDeclaration':
+							insertAfterPath = path.parentPath;
+							programPath = insertAfterPath.parentPath;
+							break;
+						default:
+							return;
+					}
+					const id = node.id;
+					if (id === null) {
+						// We don't currently handle anonymous default exports.
+						return;
+					}
+					const inferredName = id.name;
+					if (!isComponentishName(inferredName)) {
+						return;
+					}
+
+					// Make sure we're not mutating the same tree twice.
+					// This can happen if another Babel plugin replaces parents.
+					if (seenForRegistration.has(node)) {
+						return;
+					}
+					seenForRegistration.add(node);
+					// Don't mutate the tree above this point.
+
+					const handle = createRegistration(programPath, inferredName);
+					insertAfterPath.insertAfter(t.expressionStatement(t.assignmentExpression('=', handle, path.node.id)));
+				}
+			},
+			// Not possible due to "referencesImport"
+			// CallExpression(path, state) {
+			// 	if (!path.get('callee').referencesImport('preact', 'createContext')) return;
```

Comment:
We could probably replace this with `path.get('callee').node.name !== 'createContext' && !program.body.find(createContextFromPreact)`